Package:     capi4hylafax
Subject:     fix platform 64 check
Author:      Joachim Wiedorn <ad_debian at joonet.de>
Forwarded:   no
Last-Update: 2012-06-12

Fix for compiler errors on 64bit systems.

---

diff -urN s06/src/standard/aTypes.h s07/src/standard/aTypes.h
--- s06/src/standard/aTypes.h	2012-06-12 19:54:35.860990222 +0200
+++ s07/src/standard/aTypes.h	2012-06-12 22:23:40.198909661 +0200
@@ -24,8 +24,12 @@
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
 
-#if !defined C_PLATFORM_64 && defined SIZEOF_VOID_P && (SIZEOF_VOID_P == 8)
+#if defined __WORDSIZE && (__WORDSIZE == 64)
 #define C_PLATFORM_64
+#else
+#if defined SIZEOF_VOID_P && (SIZEOF_VOID_P == 8)
+#define C_PLATFORM_64
+#endif
 #endif
 
 /*---------------------------------------------------------------------------*\
@@ -48,7 +52,7 @@
 typedef unsigned int       uint32_t;
 
 #ifdef __linux__
-    #if __WORDSIZE == 64
+    #ifdef C_PLATFORM_64
         typedef long int int64_t;
         typedef unsigned long int uint64_t;
     #else
